src/Domain/SkillSetDomain.php

```diff
@@ -40,4 +40,34 @@ public function getSkillsToDisplay(): array
     {
         return \array_slice($this->getSkillSetKeys(), 0, $this->importantSkillsToDisplayLimit);
     }
+
+    public function getDependantSkillsFromSkillSet(array $skillSet): array
+    {
+        $skills = [];
+        foreach ($skillSet as $skill) {
+            $skills[] = $skill;
+            $skills = array_merge($skills, $this->getDependantSkills($skill));
+        }
+
+        return array_values(array_unique($skills));
+    }
+
+    private function getDependantSkills(string $skill): array
+    {
+        if (!\in_array($skill, $this->getSkillSetKeys(), true)) {
+            return [];
+        }
+
+        if (!\array_key_exists('includes', $this->availableSkillSets[$skill])) {
+            return [];
+        }
+
+        $dependantSkills = [];
+        foreach ($this->availableSkillSets[$skill]['includes'] as $include) {
+            $dependantSkills[] = $include;
+            $dependantSkills = array_merge($dependantSkills, $this->getDependantSkills($include));
```

Comment:
we should maybe add something to avoid infinite loops if any : we can maybe test if an included skill has already been parsed ?

Reply:
good catch :)

Reply:
WDYT if i check `availableSkillSets` integrity (ie: no infinite loop) in the constructor?

Reply:
I would have done both :-)

Reply:
is it really necessary? if the class is given a valid skills set, an infinite loop cannot happen.

Reply:
for infinite loops, you cannot trust the db if there is no validator on its side (as it can be very painful for the server ;-) )